client/allocator/src/freeing_bump.rs

```diff
@@ -314,27 +314,50 @@ impl IndexMut<Order> for FreeLists {
 	}
 }
 
+/// Memory allocation stats gathered during the lifetime of the allocator.
+#[derive(Clone, Debug, Default)]
+#[non_exhaustive]
+pub struct AllocationStats {
```

Comment:
We should also track the maximum single allocation that ever happened. Maybe we can directly track this as percentage of the maximum allowed allocation? 

```
(order.size() + HEADER_SIZE) / MAX_POSSIBLE_ALLOCATION
```